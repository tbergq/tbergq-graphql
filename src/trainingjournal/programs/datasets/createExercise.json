{
  "id": 1283,
  "set": "2",
  "reps": "2",
  "day": 201,
  "description": "2",
  "break_time": "2",
  "base_exercise": {
    "id": 10,
    "name": "Arnold Press",
    "youtube_link": "https://www.youtube.com/watch?v=6Z15_WdXmVw",
    "muscle_group": {
      "id": 4,
      "name": "Shoulders"
    },
    "description": ""
  }
}