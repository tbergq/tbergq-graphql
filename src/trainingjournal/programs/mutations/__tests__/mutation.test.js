// @flow

import { generateTestsFromFixtures } from '@kiwicom/test-utils';

import executeTestQuery from '../../../../common/services/executeTestQuery';
import createExercise from '../../datasets/createExercise.json';

fetch.mockResponses([JSON.stringify(createExercise)]);

generateTestsFromFixtures(`${__dirname}/__fixtures__`, input =>
  executeTestQuery(input),
);
