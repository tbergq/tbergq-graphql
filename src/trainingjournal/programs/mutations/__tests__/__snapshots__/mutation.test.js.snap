// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches expected output: CreateExercise.graphql 1`] = `
~~~~~~~~~~ INPUT ~~~~~~~~~~
mutation createExercise {
  createExercise(exercise: {breakTime: 1, dayId: "RGF5OjIwMQ==", description: "Nopw", reps: "2", set: "2", baseExerciseId: "YmFzZUV4ZXJjaXNlOjM4"}) {
    exerciseEdge {
      node {
        id
        name
        set
        reps
        breakTime
        baseExercise {
          id
          name
        }
      }
    }
  }
}
~~~~~~~~~~ OUTPUT ~~~~~~~~~~
{
  "data": {
    "createExercise": {
      "exerciseEdge": {
        "node": {
          "id": "RXhlcmNpc2U6MTI4Mw==",
          "name": null,
          "set": "2",
          "reps": "2",
          "breakTime": "2",
          "baseExercise": {
            "id": "QmFzZUV4ZXJjaXNlOjEw",
            "name": "Arnold Press"
          }
        }
      }
    }
  }
}
`;
